### PR TITLE
convert three digit hex to six

### PR DESCRIPTION
**Pull request by lidord-wix**
## Description
`Colors.rgba` returned wrong values when getting 3 digit hex,
I've added function to convert 3 digit hex to 6 digit hex inside `validateHex` 
Bug description: #946 

## Changelog
fix bug in `Colors.rgba`

### CODE REVIEW

File: src/style/colors.ts [hunk 1]
```diff
@@ -203,7 +203,15 @@ function validateHex(value: string) {
   if (!/^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(value)) {
     throw new Error(`${value} is invalid hex color`);
   }
-  return value.replace('#', '');
+  value = value.replace('#', '');
+  if (value.length === 3) {
+    value = threeDigitHexToSix(value);
```
Comment (ethanshar) [inline src/style/colors.ts:208]: I know using regex to duplicate the string look fancy (: 
But maybe a more readable approach will be to concatenate value with itself.

File: src/style/colors.ts [hunk 2]
```diff
@@ -203,7 +203,15 @@ function validateHex(value: string) {
   if (!/^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(value)) {
```
Comment (ethanshar) [inline src/style/colors.ts:203]: Let's add a unit test for this use case in `colors.spec.js`

File: src/style/colors.ts [hunk 3]
```diff
@@ -211,7 +211,10 @@ function validateHex(value: string) {
 }
 
 function threeDigitHexToSix(value: string) { 
-  return value.replace(/./g, '$&$&');
+  const firstDigit = value[0];
+  const secondDigit = value[1];
+  const thirdDigit = value[2];
+  return firstDigit.concat(firstDigit, secondDigit, secondDigit, thirdDigit, thirdDigit);
```
Comment (ethanshar) [inline src/style/colors.ts]: Why not just return `value + value` ?

Comment (lidord-wix) [reply src/style/colors.ts]: We need to double each letter: 
`threeDigitHexToSix(123)` should return `112233` and not `123123`

Comment (ethanshar) [reply src/style/colors.ts]: You are completely right! my bad :( 
In that case the regex solution is definitely preferable... 
apologies for the trouble

**Review by ethanshar — APPROVED** [review]

**Pull request merged**